src/Mod/PartDesign/Gui/TaskHoleParameters.cpp

```diff
@@ -75,11 +75,20 @@ TaskHoleParameters::TaskHoleParameters(ViewProviderHole* HoleView, QWidget* pare
     auto pcHole = getObject<PartDesign::Hole>();
     bool isNone = std::string(pcHole->ThreadType.getValueAsString()) == "None";
 
-    ui->Threaded->setHidden(isNone);
+    ui->labelHoleType->setHidden(isNone);
+    ui->HoleType->setHidden(isNone);
     ui->ThreadSize->setHidden(isNone);
     ui->labelSize->setHidden(isNone);
 
-    ui->Threaded->setChecked(pcHole->Threaded.getValue());
+    if (pcHole->Threaded.getValue()) {
+        if (pcHole->ModelThread.getValue()) {
+            ui->HoleType->setCurrentIndex(static_cast<int>(HoleTypeIndex::ModeledThread));
+        } else {
+            ui->HoleType->setCurrentIndex(static_cast<int>(HoleTypeIndex::TapDrill));
+        }
+    } else {
+        ui->HoleType->setCurrentIndex(static_cast<int>(HoleTypeIndex::Clearance));
+    }

```

Comment:
Can it be simplified if it is extracted to separate function? I'm thinking something like this:

```
auto getHoleType = [](PartDesign::Hole* pcHole) {
    if (pcHole->Threaded.getValue()) {
        return pcHole->ModelThread.getValue() ? HoleTypeIndex::ModeledThread : HoleTypeIndex::TapDrill;
    }
    
    return HoleTypeIndex::Clearance;
};

ui->HoleType->setCurrentIndex(getHoleType(pcHole));
```